### ISSUE TITLE
Scheduling pop-up freezes when user is in the middle of typing a date

### ISSUE THREAD

**Issue by jikamens** [Extended-Thunder/send-later#625]
A user reports:

>When I write an email and click on the "Send Later" button, a small window opens. When I manually enter a date in the first line in the "European" format (d[d].m[m].[yy]yy), Send Later freezes when I enter the 2 from yy[yy]. Afterwards, Thunderbird can still be used, but "Send Later" has to be restarted.
>The error does not occur with day or month.
>That is not a big problem either. I can change the date in the second line without any problems. Once there is something in the first line, I can also change the numbers there. But only one at a time. If I delete "202" from "2023", for example, Send Later freezes again.

this is being cuased by `estimateSentTime` in static.js:

```
  estimateSendTime(scheduledDate, previousLoop, loopMinutes) {
    SLStatic.trace(
      `estimateSendTime(${scheduledDate}, ${previousLoop}, ${loopMinutes})`,
    );
    scheduledDate = SLStatic.convertDate(scheduledDate);
    scheduledDate.setSeconds(0);

    let estimate = new Date(previousLoop);
    while (
      estimate.getTime() < scheduledDate.getTime() ||
      estimate.getTime() < Date.now() + 1000
    ) {
      estimate = new Date(estimate.getTime() + 60000 * loopMinutes);
    }

    return estimate;
  },
```

That's a tight `while` loop moving forward one minute at a time from early in the previous millennium to today!

We should probably check if the date passed in more than `loopMinutes` in the past and if so then just return that date and let the calling code handle the fact that it's in the past (which I believe it already does).

**Comment by jikamens** [comment]
Hmm, nope, I'm wrong about the root cause of the freezing. I read the code above wrong, I think. Need to think about this more.

**Comment by jikamens** [comment]
I was close. The problem isn't that the loop is counting from far into the past to now, the problem is that the loop is counting from now until far into the future. 9/1/2 (US) e.g. is being parsed as 9/1/2102.

**Comment by jikamens** [comment]
This issue should be fixed in release 10.1.5 of Send Later. To check what version of Send Later you have:

* Open the Add-ons page in Thunderbird ("Add-ons and Themes" from the corner "hamburger" menu).
* Click on Extensions on the left if you're not already on the left if you're not already there.
* Find Send Later in the list of extensions and click on it to open its details page.
* The version number should be displayed there next to
"Version".

This release is not yet available for download from addons.thunderbird.net, but you can download and install it from its [GitHub release page](https://github.com/Extended-Thunder/send-later/releases/tag/v10.1.5) as follows:

* Download [this file](https://github.com/Extended-Thunder/send-later/releases/download/v10.1.5/send_later.xpi) to your computer. (If you are using Firefox, make sure to right click on the link and select
"Save Link As..." rather than just clicking on it, because otherwise Firefox will try to install it as a Firefox add-on.)
* Open the Add-ons page in Thunderbird ("Add-ons and Themes" from the corner "hamburger" menu).
* Click the gear icon at the top and select "Install Add-on From File...".
* Browse to and select the downloaded file.
* Click through the installation pop-ups.
* You can delete the downloaded file once you've installed it into Thunderbird.

If you prefer to wait, the release will eventually be available on addons.thunderbird.net, and unless you've turned off automatic add-on updates, Thunderbird will eventually update to the new version automatically. To update immediately once it's available:

* Go to [addons.thunderbird.net](https://addons.thunderbird.net/thunderbird/addon/send-later-3/versions/) and confirm that release 10.1.5 is available for download there.
* Open the Add-ons page in Thunderbird ("Add-ons and Themes" from the corner "hamburger" menu).
* Click on Send later.
* Click the gear icon and select "Check for Updates".
* The add-on should update and then you should see the message, "Your add-ons have been updated," next to the gear icon.

Please let me know if you still see this issue in this release.
